### PR TITLE
[Chassis] [BGP templates] Enhancement for not to send given route to APP_DB and also fallback to default route

### PR DESCRIPTION
**Pull request by abdosi**
What I did:

- Enhancement of [VOQ Chassis i-BGP Templates]

1. On Upstream LC deny routes that are ok to fallback to Default Routes and have community of DEVICE_INTERNAL_COMMUNITY  (Routes learnt from AZNG neighbors: https://github.com/sonic-net/sonic-mgmt/blob/e4ad13a90450c930d272b13a8b1e49a66284b036/ansible/templates/minigraph_png.j2#L212)
2. On Downstream LC allow routes that are ok to fallback to Default Routes  and have community of DEVICE_INTERNAL_COMMUNITY set a special tag for those.

- Enhancement of [Packet Chassis i-BGP internal Templates]

1. On Upstream LC allow routes that are ok to fallback to Default Routes and have community of DEVICE_INTERNAL_COMMUNITY (Routes learnt from AZNG neighbors: https://github.com/sonic-net/sonic-mgmt/blob/e4ad13a90450c930d272b13a8b1e49a66284b036/ansible/templates/minigraph_png.j2#L212)
2. On Downstream LC deny routes that are ok to fallback to Default Routes and   have community of DEVICE_INTERNAL_COMMUNITY set a special tag for those.

- Enhancement of general templates for Chassis when running on Upstream LC
1.  Allow Default Route always
2. Any other route mark 
         a. On chassis-packet set tag as eligible for default fallback and set community of DEVICE_INTERNAL_COMMUNITY
         b. On Voq chassis set tag as eligible for not to send to app db and set community of DEVICE_INTERNAL_COMMUNITY

How I verify:
Update UT.
Manual Verification.

### CODE REVIEW

File: dockers/docker-fpm-frr/frr/bgpd/templates/voq_chassis/policies.conf.j2 [hunk 1]
```diff
@@ -12,6 +13,17 @@ route-map FROM_VOQ_CHASSIS_V4_PEER permit 1
 route-map FROM_VOQ_CHASSIS_V4_PEER permit 2
   match community NO_EXPORT
   set local-preference 80
+  on-match next
+!
+{% if CONFIG_DB__DEVICE_METADATA['localhost']['subtype'] == 'UpstreamLC' %}
```
Comment (arlakshm) [inline dockers/docker-fpm-frr/frr/bgpd/templates/voq_chassis/policies.conf.j2:18]: is this correct, should it be `DownstreamLC` ?

File: dockers/docker-fpm-frr/frr/bgpd/templates/general/policies.conf.j2 [hunk 2]
```diff
@@ -35,13 +37,44 @@ route-map FROM_BGP_PEER_V4 permit 10
 !
 route-map FROM_BGP_PEER_V4 permit 11
   match community allow_list_default_community
+{% if CONFIG_DB__DEVICE_METADATA and 'localhost' in CONFIG_DB__DEVICE_METADATA and 'type' in CONFIG_DB__DEVICE_METADATA['localhost'] and 'subtype' in CONFIG_DB__DEVICE_METADATA['localhost'] %}
+{% if CONFIG_DB__DEVICE_METADATA['localhost']['type'] == 'SpineRouter' and CONFIG_DB__DEVICE_METADATA['localhost']['subtype'] == 'UpstreamLC' %}
```
Comment (arlakshm) [inline dockers/docker-fpm-frr/frr/bgpd/templates/general/policies.conf.j2:41]: Do we need to add dependencies for these in bgpcfgd?

File: dockers/docker-fpm-frr/frr/bgpd/templates/voq_chassis/policies.conf.j2 (hunk 1 above)
Comment (abdosi) [reply dockers/docker-fpm-frr/frr/bgpd/templates/voq_chassis/policies.conf.j2:18]: yes, this condition is to check if any azng route (eligible for fallback)  learnt via iBGP on upstream LC should be deny

File: dockers/docker-fpm-frr/frr/bgpd/templates/general/policies.conf.j2 (hunk 2 above)
Comment (abdosi) [reply dockers/docker-fpm-frr/frr/bgpd/templates/general/policies.conf.j2:41]: we already have dependency for `type` which i think should be good for `subtype` also. As this data is written at same time to config db.

File: dockers/docker-fpm-frr/frr/bgpd/templates/voq_chassis/policies.conf.j2 (hunk 1 above)
Comment (abdosi) [reply dockers/docker-fpm-frr/frr/bgpd/templates/voq_chassis/policies.conf.j2:18]: > yes, this condition is to check if any azng route (eligible for fallback) learnt via iBGP on upstream LC should be deny

**Review by arlakshm — APPROVED** [review]

**Comment by yejianquan** [comment]
@bingwang-ms could you help to approve cherry-pick?

**Comment by yejianquan** [comment]
> @bingwang-ms could you help to approve cherry-pick?

Added to the wish list

**Comment by bingwang-ms** [comment]
Hi @StormLiangMS, can you please help confirm this change will not impact Non-Chassis platform?

**Pull request merged**